qos-service-impl/src/main/java/com/palantir/atlasdb/qos/QosResource.java

```diff
@@ -18,27 +18,42 @@
 import com.palantir.atlasdb.qos.com.palantir.atlasdb.qos.agent.QosClientConfigLoader;
 import com.palantir.atlasdb.qos.config.QosClientLimitsConfig;
 import com.palantir.atlasdb.qos.ratelimit.ClientLimitMultiplier;
+import com.palantir.atlasdb.util.MetricsManager;
 
 public class QosResource implements QosService {
     private final QosClientConfigLoader qosClientConfigLoader;
     private final ClientLimitMultiplier clientLimitMultiplier;
+    private static volatile double readLimitMultiplier = 1.0;
+    private static volatile double writeLimitMultiplier = 1.0;
 
     public QosResource(QosClientConfigLoader qosClientConfigLoader, ClientLimitMultiplier clientLimitMultiplier) {
         this.qosClientConfigLoader = qosClientConfigLoader;
         this.clientLimitMultiplier = clientLimitMultiplier;
+        new MetricsManager().registerMetric(QosResource.class, "readLimitMultiplier", this::getReadLimitMultiplier);
+        new MetricsManager().registerMetric(QosResource.class, "writeLimitMultiplier", this::getWriteLimitMultiplier);
     }
 
     @Override
     public long readLimit(String client) {
         QosClientLimitsConfig qosClientLimitsConfig = qosClientConfigLoader.getConfigForClient(client);
-        return (long) (clientLimitMultiplier.getClientLimitMultiplier(qosClientLimitsConfig.clientPriority())
-                * qosClientLimitsConfig.limits().readBytesPerSecond());
+        readLimitMultiplier = this.clientLimitMultiplier.getClientLimitMultiplier(
```

Comment:
This is awkward with the current `CassandraMetricsClientLimitMultiplier` implementation, since any high priority clients will update the limit back to 1.0. We'll likely see some sawtooth metrics if there are clients at different priority levels while throttling is ongoing.